File: FTC23/src/main/java/org/firstinspires/ftc/teamcode/modules/hardware/Module.java
package org.firstinspires.ftc.teamcode.modules.hardware;

import com.qualcomm.robotcore.hardware.HardwareMap;


abstract class Module {
    protected HardwareMap hardwareMap = null;

    public Module(HardwareMap hMap){
        hardwareMap = hMap;
    }

    /**
     * Initțializează modulul
     */
    abstract void initModule();

    abstract void updateModule();
}
 

File: FTC23K/src/main/java/org/firstinspires/ftc/teamcode/modules/hardware/Module.kt
package org.firstinspires.ftc.teamcode.modules.hardware

import com.qualcomm.robotcore.hardware.HardwareMap

abstract class Module(hMap: HardwareMap?) {
    @JvmField
    protected var hardwareMap: HardwareMap? = null

    init {
        if (hMap != null) {
            hardwareMap = hMap
        }
    }

    abstract fun initModule()
    abstract fun updateModule()
}